compiler/luci-interpreter/src/kernels/Conv2D.cpp

```diff
@@ -87,6 +88,22 @@ void Conv2D::configure()
                                   filter_width, output_width);
 
   _output->resize({batches, output_height, output_width, output_depth});
+
+  // Allocate tensor for Im2Col, if needed.
+  // The checks here should be aligned with the actual implementation.
+  const bool need_dilated_im2col =
+      _params.dilation_height_factor != 1 || _params.dilation_width_factor != 1;
+  const bool need_non_dilated_im2col = _params.stride_height != 1 || _params.stride_width != 1 ||
+                                       filter_height != 1 || filter_width != 1;
+  const bool need_im2col = need_dilated_im2col || need_non_dilated_im2col;
```

Comment:
I see there are bunch of additional conditions in tflite:
```
  const bool is_hybrid_non_dilated = is_hybrid && need_non_dilated_im2col;
  const bool is_quantized =
      input->type == kTfLiteUInt8 || input->type == kTfLiteInt8;

  switch (kernel_type) {
    case kReference:
      if (is_hybrid) {
        return true;
      } else {
        return false;
      }    
    case kGenericOptimized:
    case kCblasOptimized:
      if (is_hybrid && !need_non_dilated_im2col) {
        return false;
      } else {
        return true;
      }    
    case kMultithreadOptimized:
      if (is_hybrid_non_dilated || is_quantized ||
          !data->supports_multithreaded_kernel) {
        return true;
      } else {
        return false;
      }    
    default:
      return false;
  }
```
maybe we should include them (at least kMultithreadOptimized branch) here?

Reply:
Those checks are for "modern" kernels. The checks I added are based on "legacy" kernels.

Reply:
P.S. I also had concerns about im2col buffer size in case of dilated/normal convolution, but looks like there are no poblems